build.psm1

```diff
@@ -2179,6 +2179,10 @@ function New-MSIPackage
         $WiXHeatLog   | Out-String | Write-Verbose -Verbose
         $WiXCandleLog | Out-String | Write-Verbose -Verbose
         $WiXLightLog  | Out-String | Write-Verbose -Verbose
+        if ($null -ne $env:CI)
+        {
+           $host.SetShouldExit(-1)
```

Comment:
Instead of having the CmdLet alter the host behavior, AppVeyor.psm1 should detect this error and fail the CI.  This might become unreliable if we attempt this is multiple places.

Another option, is detect we are in AppVeyor, and [Add a compilation message](https://www.appveyor.com/docs/build-worker-api/#add-compilation-message)

Reply:
I applied your suggestion but the error message from Appveyor is very obscure then: https://ci.appveyor.com/project/PowerShell/powershell/build/1.0.7407